docs/guides/advanced/array-types.md

```diff
@@ -0,0 +1,25 @@
+---
+id: array-types
+title: 'Array Types & Fields'
+---
+
+It is much easier to declare the array field's type as `type[]` instead of `Array<type>`.
+
+But in some cases, Typescript could give you a warning, when you would like to use [any mongoose array methods](https://mongoosejs.com/docs/api/array.html) on the array field.
+To avoid such behavior, you could always declare the array field via `mongoose.Types.Array<type>` or `mongoose.Schema.Types.Array<type>`
+
+Example with `Class`:
+
+```ts
+class Model {
+  //required field, with emplty array by default.
+  @prop({ required: true, default: [] })
+  public field!: mongoose.Types.Array<string>;
+}
+```
+
+## Why does it could happen?
```

Comment:
is there an better wording than this?

Reply:
Well, to be honest, I don't have an alternative. I could propose to remove the `h2` title at this phrase doesn't suit the docs.

Reply:
```suggestion
## Why is the long type needed?
```